src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -548,16 +548,26 @@ private async Task<string> GetDurableMajorVersionAsync(JObject hostJson, ScriptJ
             bool isUsingBundles = hostJson != null && hostJson.TryGetValue("extensionBundle", StringComparison.OrdinalIgnoreCase, out _);
             if (isUsingBundles)
             {
-                // TODO: As of 2019-12-12, there are no extension bundles for version 2.x of Durable.
-                // This may change in the future.
-                return "1";
+                // From Functions runtime V4 onwards, only bundles >= V2.x is supported, which implies the app should be using DF V2 or greater.
+                return "2";
             }
 
+            // There's two places where `extensions.json` may be located
```

Comment:
The flow here should match what is used for the extension probing, using the same order of precedence:

1 - `bin` folder
2 - script root
3 - System path (`.azurefunctions`) which is the constant `AzureFunctionsSystemDirectoryName` inside of the script root path.

You can reference the logic here:
- https://github.com/Azure/azure-functions-host/blob/66731fa6715f02610c3d0842c56488e540e695e3/src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs#L128-L154